src/DrawingSurface.java

```diff
@@ -0,0 +1,330 @@
+
+					if (doorsRect.get(i).contains(player.getX(), player.getY())
+							|| doorsRect.get(i).contains(player.getX() + player.getWidth(), player.getY())
+							|| doorsRect.get(i).contains(player.getX(), player.getY() + player.getHeight())
+							|| doorsRect.get(i).contains(player.getX() + player.getWidth(),
+									player.getY() + player.getHeight())) {
+
+						if (classroom.get(i).isFinished() == false) {
+							setting = i;
+							player.setX(classroom.get(i).startPointX());
+							player.setY(classroom.get(i).startPointY());
+
+						}
+
+					}
+				}
+			}
+
+		} else if (setting >= 0) {
+
+			classroom.get(setting).draw(this);
+			player.draw(this);
+			if (firstSetNot0) {
+				for (int i = 0; i < zombies.size(); i++) {
+					zombies.get(i).makeHidden();
+					zombies.get(i).setX(40);
+					zombies.get(i).setY(100 * i);
+					firstSetNot0 = false;
+				}
+			}
+
+			for (int i = 0; i < zombies.size(); i++) {
+				if (zombies.get(i).isShown()) {
+					zombies.get(i).draw(this, player);
+				}
+
+			}
+
+			ArrayList<Point> answerCoord = classroom.get(setting).answerLocations();
+			ArrayList<Rectangle> answerRect = new ArrayList<Rectangle>(answerCoord.size());
+			for (int i = 0; i < answerCoord.size(); i++) {
+				answerRect.add(new Rectangle(answerCoord.get(i).x - sizeAnswers * 2,
+						answerCoord.get(i).y - sizeAnswers * 1, sizeAnswers * 4, sizeAnswers * 2));
+			}
+			for (int i = 0; i < answerCoord.size(); i++) {
+				if (answerRect.get(i).contains((float) (player.getX() + 32), (float) player.getY() + 32)
+						|| answerRect.get(i)
+								.contains(player.getX() + 32
+										+ (((float) (player.getX() + player.getWidth() / 2) + 32)
+												- ((float) player.getX() + 32)),
+										(float) (player.getY() + 32))
+						|| answerRect.get(i).contains((float) (player.getX() + 32),
+								(float) player.getY() + 32
+										+ (((float) (player.getY() + player.getHeight() / 2) + 32)
+												- ((float) player.getY() + 32)))
+						|| answerRect.get(i).contains(
+								(float) (player.getX() + 32
+										+ (((float) (player.getX() + player.getWidth() / 2) + 32)
+												- ((float) player.getX() + 32))),
+								(float) player.getY() + 32 + (((float) (player.getY() + player.getHeight() / 2) + 32)
+										- ((float) player.getY() + 32)))) {
+
+					if (classroom.get(setting).getCorrectAnswer() == i) {
+						classroom.get(setting).changeClassToFinished();
+						setting = -3;
+						firstSetNot0 = true;
+						player.setX(map.returnStartPointX());
+						player.setY(map.returnStartPointY());
+
+					} else if (classroom.get(setting).getLives() == 1) {
+						classroom.get(setting).removeLives();
+						setting = -1;
+					} else if (classroom.get(setting).getLives() == 2) {
+						player.setX(classroom.get(i).startPointX());
+						player.setY(classroom.get(i).startPointY());
+						classroom.get(setting).removeLives();
+						classroom.get(setting).removeAnswer(i);
+					}
```

Comment:
1) Move this code to Classroom and let it handle collisions with answers.
2) Create ArrayLists (like answerRects) in advance rather than for each frame of drawing.
3) Consider literally using answerRects to actually draw the answer rectangles - this way you know they're accurate.
4) Rectangle class has a intersects() method that takes in another Rectangle. Simplify collision detection like answerRect.get(i).intersects(new Rectangle(player.x,player.y,player.width,player.height))
5) Add a println to each of the if-statements to know which one you're hitting (or hitting none of them) for debugging.